core/src/main/java/com/kin/ecosystem/core/Configuration.java

```diff
@@ -57,20 +63,29 @@ public static ApiClient getDefaultApiClient() {
 					@Override
 					public Response intercept(Chain chain) throws IOException {
 						Request originalRequest = chain.request();
-
 						final String path = originalRequest.url().encodedPath();
-						AuthToken authToken = null;
-						if (!path.equals(USERS_PATH)) {
-							authToken = AuthRepository.getInstance().getAuthTokenSync();
-						}
 
-						if (authToken != null) {
-							Request authorisedRequest = originalRequest.newBuilder()
-								.header(AUTHORIZATION, BEARER + authToken.getToken())
-								.build();
-							return chain.proceed(authorisedRequest);
-						} else {
+						if (path.equals(USERS_PATH)) {
 							return chain.proceed(originalRequest);
+						} else {
+							AuthToken authToken = AuthRepository.getInstance().getAuthTokenSync();
+							if (authToken != null) {
+								Request authorisedRequest = originalRequest.newBuilder()
+									.header(AUTHORIZATION, BEARER + authToken.getToken())
+									.build();
+								return chain.proceed(authorisedRequest);
+							} else {
+								// Stop the request from being executed.
+								Logger.log(new Log().withTag("ApiClient").text("No token - response error on client"));
+								return new Response.Builder()
+									.code(NO_TOKEN_ERROR_CODE)
```

Comment:
you can move the error response response builded to a method for better readability (IMO)